src/rdydata.c

```diff
@@ -181,6 +190,43 @@ PetscErrorCode RDyGetLocalCellYMomentums(RDy rdy, const PetscInt size, PetscReal
   PetscFunctionReturn(PETSC_SUCCESS);
 }
 
+PetscErrorCode RDySetWaterSourceForRegion(RDy rdy, const PetscInt region_idx, PetscReal value) {
```

Comment:
Should we consider region-wise source operators? Can the source terms be different for each region?

Reply:
For Dongyu's simulation configuration, we must apply six different source terms to six regions. So, we need region-specific source terms.